RELEASE.md

```diff
@@ -1,64 +1,224 @@
 
-- Are there any critical issues observed?
+#### All-in-one script (for example version `v1.1.0`, commit `8ab7cef`):
 
-  - Examples: Memory leaks, abnormally high memory usage, abnormally high CPU performance, etc.
+- The team selects a commit from `rc/v1.1.0` as a commit to tag and publish.
+- `yarn release:tag-rc 1.1.0 8ab7cef`
+  - Must be run locally from a write-access account capable of triggering CI.
+  - This script may alternatively be run on the the checked out `HEAD`:
+    - `git checkout 8ab7cef`
+    - `yarn release:tag-rc 1.1.0`
 
-- Has profitibility been affected?
+#### Manual steps (for example version `v1.1.0`, commit `8ab7cef`):
 
-  - Has profitability decreased and whether or not the variance is acceptable.
+- Check out the commit:
+  - `git checkout 8ab7cef`
+- Tag resulting commit as `v1.1.0-rc.0` with an annotated tag, and push the tag.
+  - `git tag -am "v1.1.0-rc.0" v1.1.0-rc.0`
+  - `git push origin v1.1.0-rc.0`
+
+### 3. Test release candidate
+
+Once a release candidate is created, the Lodestar team begins a 3 day testing period.
+
+If there is a bug discovered during the testing period which significantly impacts performance, security, or stability, and it is determined that it is no longer prudent to promote the `rc.x` candidate to `stable`, then it will await a bug fix by the team. The fix will be committed, maybe back-ported to `unstable` and we publish and promote the new commit to `rc.x+1`. The 3 day testing period will reset.
+
+For example: After 3-5 days of testing, is performance equal to or better than latest stable?
+
+- **Yes**: Continue to the next release step
+- **No**: If it a small issue fixable quickly (hot-fix)?
+  - **Yes**: push fixes to branch, go to step 2, incrementing the rc version
+  - **No**: abort the release. Close the `v1.1.0 release` PR, delete the branch, and start the whole release process over.
+
+### 4. Merge release candidate
+
+- Ensure step 2 testing is successful and there is sufficient consensus to release `v1.1.0`.
+- Approving the `v1.1.0 release` PR means a team member marks the release as safe, after personally reviewing and / or testing it.
+- Merge `v1.1.0 release` PR to stable **with "merge commit"** strategy to preserve all history.
+- Merge stable `stable` into `unstable` **with merge commit** strategy. If that's not possible, open a PR and handle conflicts latter.
+
+### 5. Tag stable release
+
+Tagging a stable release will trigger CI to publish to NPM, dockerhub, and Github releases.
+
+#### All-in-one script (for example version `v1.1.0`):
+
+- `git checkout stable`
+- `yarn release:tag-stable 1.1.0`
+  - Must be run locally from a write-access account capable of triggering CI.
+#### Manual steps (for example version `v1.1.0`):
+
+- Check out the new stable
+  - `git checkout stable`
+- Tag it as `v1.1.0` with an annotated tag, push commit and tag.
+  - `git tag -am "v1.1.0" v1.1.0`
+  - `git push --tag`
```

Comment:
@wemeetagain we should never push all tags blindly. If you had stale tags (already removed in remote) it could create chaos by triggering many releases at once. it's best to push a single tag only with `git push origin refs/tags/v1.1.0`